src/storage/blocksstable/ob_macro_block.cpp

```diff
@@ -265,6 +280,7 @@ void ObDataStoreDesc::reset()
   major_working_cluster_version_ = 0;
   iter_complement_ = false;
   is_unique_index_ = false;
+  allocator_.reset();
```

Comment:
it's risky here,  prev codes just reset some state and not release memory from allocator, should check whether some pointers are still used after reset

Reply:
In the original code, I find that the allocator_ is not used in ObDataStoreDesc functions. But now I use it to allocate memory for column_ids_... So I think it should be reset in the reset() function.